the-new-developers/gatehopper
# Added preloader to avoid network calls and computationally expensive scaling between scenes

## Changes:
Added a preload function to scenes.
This is meant to load all assets into memory before the program runs.

Closes #48 

## How to use
In a given scene class:
- Write a preload function
- have the function return an array of promises

## How to test this PR:
1. Run the site
2. Open the network profiler (typically Ctrl+Shift+i)
3. Refresh the page

If this solution does as I say it does, all assets for the cheatsheet and challenge mode pages should not be loaded on scene switch, but instead, on page load.

## Checklist:
